pkg/sentry/kernel/task_start.go

```diff
@@ -251,6 +251,18 @@ func (ns *PIDNamespace) allocateTID() (ThreadID, error) {
 		// Is it available?
 		_, ok := ns.tasks[tid]
 		if !ok {
+			// When leader of process group or session exit, the
+			// process group ID or session ID can not be reused.
+			// Skip the tid if it is using by a process group or
+			// session when allocating a new tid.
+			if _, ok := ns.processGroups[ProcessGroupID(tid)]; ok {
+				continue;
```

Comment:
`continue` here skips the `tid == ns.last` check below. IMO the cleanest approach would be:

```
		// Is it available?
		tidInUse := func() bool {
			if _, ok := ns.tasks[tid]; ok {
				return true
			}
			if _, ok := ns.processGroups[ProcessGroupID(tid)]; ok {
				return true
			}
			if _, ok := ns.sessions[SessionID(tid)]; ok {
				return true
			}
			return false
		}()
		if !tidInUse {
			ns.last = tid
			return tid, nil
		}

		// Did we do a full cycle?
```

Reply:
Thanks, your revise looks better. I had push another update, could help to review again, thanks.